### PR TITLE
#208 adds signaling API

### PR DESCRIPTION
**Pull request by msach22**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -204,7 +204,40 @@ use OpenTok\OpenTok;
...
+`OpenTok\OpenTok` class. The `$options` parameter is an associative array used to set the data, type, and connection Id. Please keep in mind
+that the `connectionId` is an optional parameter and should be to used to send a signal to a specific connection.
+
+```php
+use OpenTok\OpenTok;
+
+// Send a signal to a specific connection
+
+$signalOptions = array(
+    'connectionId' => 'da9cb410-e29b-4c2d-ab9e-fe65bf83fcaf',
```
Comment (jeffswartz) [inline README.md]: Please revise the based on the discussion in https://github.com/opentok/OpenTok-PHP-SDK/issues/208. I think we are deciding on OpenTok.signal($sessionId, $payload, $connectionId).

File: src/OpenTok/Exception/SignalDomainException.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+namespace OpenTok\Exception;
+
+// TODO: this kind of exception has a detailed message from the HTTP response, use it.
+/**
+* Defines an exception thrown when a call to a signal method results in an error response from
+* the server.
+*/
+class SignalDomainException extends \OpenTok\Exception\DomainException implements \OpenTok\Exception\SignalException
+{
+}
```
Comment (jeffswartz) [inline src/OpenTok/Exception/SignalDomainException.php]: I think we should add exception classes for the following:

* SignalConnectionException -- The client specified by the connectionId is not connected to the session.
* SignalLengthException (or some better name) -- The type string exceeds the maximum length (128 bytes), or the data string exceeds the maximum size (8 kB). (Or should SignalUnexpectedValueException cover this?)

File: tests/OpenTok/OpenTokTest.php [hunk 3]
```diff
@@ -1442,5 +1442,92 @@ public function testSipCallFrom()
...
+        $this->assertEquals(true, TestHelpers::validateOpenTokAuthHeader($this->API_KEY, $this->API_SECRET, $authString));
+
+        // TODO: test the dynamically built User Agent string
+        $userAgent = $request->getHeaderLine('User-Agent');
+        $this->assertNotEmpty($userAgent);
+        $this->assertStringStartsWith('OpenTok-PHP-SDK/4.1.1-alpha.1', $userAgent);
+
+        $body = json_decode($request->getBody());
+        $this->assertEquals('random message', $body->data);
+
```
Comment (jeffswartz) [inline tests/OpenTok/OpenTokTest.php]: Add tests for all error conditions.

File: src/OpenTok/Exception/SignalDomainException.php (hunk 2 above)
Comment (msach22) [reply src/OpenTok/Exception/SignalDomainException.php]: @jeffswartz I like the idea of `SignalConnectionException`, but for the second one, I think we can have `SignalUnexpectedValueException` which would catch the properties and maximum length

**Pull request merged**